sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/BodyPublisherUtils.java

```diff
@@ -0,0 +1,149 @@
+import com.azure.core.implementation.util.StringContent;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.ProgressReporter;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.adapter.JdkFlowAdapter;
+import reactor.core.publisher.Flux;
+
+import java.io.FileNotFoundException;
+import java.io.UncheckedIOException;
+import java.net.http.HttpRequest;
+import java.nio.ByteBuffer;
+import java.util.concurrent.Flow;
+
+import static java.net.http.HttpRequest.BodyPublishers.fromPublisher;
+import static java.net.http.HttpRequest.BodyPublishers.noBody;
+import static java.net.http.HttpRequest.BodyPublishers.ofByteArray;
+import static java.net.http.HttpRequest.BodyPublishers.ofFile;
+import static java.net.http.HttpRequest.BodyPublishers.ofInputStream;
+import static java.net.http.HttpRequest.BodyPublishers.ofString;
+
+final class BodyPublisherUtils {
+    private static final ClientLogger LOGGER = new ClientLogger(BodyPublisherUtils.class);
+
+    private BodyPublisherUtils() {
+    }
+
+    /**
+     * Creates BodyPublisher depending on underlying body content type.
+     * If progress reporter is not null, configures it to track request body upload.
+     *
+     * @param request {@link com.azure.core.http.HttpRequest} instance
+     * @progressReporter optional progress reporter.
+     * @return the request BodyPublisher
+     */
+    public static HttpRequest.BodyPublisher toBodyPublisher(com.azure.core.http.HttpRequest request, ProgressReporter progressReporter) {
+        BinaryData body = request.getBodyAsBinaryData();
+        if (body != null) {
+            BinaryDataContent bodyContent = BinaryDataHelper.getContent(body);
+            if (bodyContent instanceof ByteArrayContent) {
+                return getPublisherWithReporter(ofByteArray(bodyContent.toBytes()), progressReporter);
+            } else if (bodyContent instanceof StringContent || bodyContent instanceof SerializableContent) {
+                return getPublisherWithReporter(ofString(bodyContent.toString()), progressReporter);
+            } else if (bodyContent instanceof FileContent) {
+                try {
+                    return getPublisherWithReporter(ofFile(((FileContent) bodyContent).getFile()), progressReporter);
+                } catch (FileNotFoundException e) {
+                    throw LOGGER.logExceptionAsError(new UncheckedIOException(e));
+                }
+            } else if (bodyContent instanceof InputStreamContent) {
+                return getPublisherWithReporter(ofInputStream(bodyContent::toStream), progressReporter);
+            } else {
+                final String contentLength = request.getHeaders().getValue("content-length");
+                return getPublisherWithReporter(toBodyPublisher(request.getBody(), contentLength), progressReporter);
+            }
+        } else {
+            return noBody();
+        }
```

Comment:
Minor nit: Thoughts on flipping the if/else here so `body == null` can be an early out and we remove a level of nesting if/else